app/assets/v2/js/grants/new.js

```diff
@@ -0,0 +1,143 @@
+        $('#js-drop input').css('visible', 'invisible');
+        $('#js-drop').css('padding', 0);
+      };
+      reader.readAsDataURL(this.files[0]);
+    }
+  });
+
+  $('.js-select2, #frequency_unit').each(function() {
+    $(this).select2();
+  });
+
+  $('#create-grant').validate({
+    submitHandler: function(form) {
+      var data = {};
+      var disabled = $(form)
+        .find(':input:disabled')
+        .removeAttr('disabled');
+
+      $.each($(form).serializeArray(), function() {
+        data[this.name] = this.value;
+      });
+
+      $('#token_symbol').val($('#js-token option:selected').text());
+
+      console.log('compiledSubscription', compiledSubscription);
+
+      // Begin New Deploy Subscription Contract
+      let SubscriptionContract = new web3.eth.Contract(compiledSubscription.abi);
+
+      // These args are baseline requirements for the contract set by the sender. Will set most to zero to abstract complexity from user.
+      let args = [
+        // admin_address
+        data.admin_address,
+        // required token
+        data.denomination,
+        // required tokenAmount
+        web3.utils.toTwosComplement(0),
+        // data.frequency
+        web3.utils.toTwosComplement(0),
+        // data.gas_price
+        web3.utils.toTwosComplement(0),
+        // contract version
+        web3.utils.toTwosComplement(0)
+      ];
+
+      console.log('args', args);
+
+      web3.eth.getAccounts(function(err, accounts) {
+        web3.eth.net.getId(function(err, network) {
+          SubscriptionContract.deploy({
+            data: compiledSubscription.bytecode,
+            arguments: args
+          }).send({
+            from: accounts[0],
+            gas: 3000000,
+            gasPrice: 4000000000
+          }).on('error', function(error) {
+            console.log('1', error);
+            alert('Your contract failed to deploy. Please try again.');
+          }).on('transactionHash', function(transactionHash) {
+            console.log('2', transactionHash);
+            $('#transaction_hash').val(transactionHash);
+            document.issueURL = $('#input-url').val();
+            enableWaitState('#new-grant');
+
+            var callFunctionWhenTransactionMined = function(transactionHash) {
+              web3.eth.getTransactionReceipt(transactionHash, function(error, result) {
+                if (result) {
+                  $('#contract_address').val(result.contractAddress);
+                  $.each($(form).serializeArray(), function() {
+                    data[this.name] = this.value;
+                  });
+                  form.submit();
+                } else {
+                  setTimeout(function() {
+                    callFunctionWhenTransactionMined(transactionHash);
+                  }, 1000);
+                }
+              });
+            };
+
+            callFunctionWhenTransactionMined(transactionHash);
+          });
+          // .on('receipt', function(receipt) {
```

Comment:
@owocki This is the code the should be reimplemented when the intermittent contract deploy issue is fixed with web3.js. It was left commented out for this reason instead of removing. Should we remove it?